fluent-syntax/src/parser.js

```diff
@@ -59,9 +59,32 @@ export default class FluentParser {
 
     while (ps.current()) {
       const entry = this.getEntryOrJunk(ps);
+      const blankLines = ps.skipBlankLines();
+
+      // Regular Comments require special logic. Comments may be attached to
+      // Messages or Terms if they are followed immediately by them. However
+      // they should parse as standalone when they're followed by Junk.
+      // Consequently, we only attach them once we know that the Message or the
+      // Term parsed successfully.
```

Comment:
Not sure if a logic based on `lastComment` would be easier or not. I.e., if you hit a `Comment` without blankLines, stash it. If you hit a `Message` or `Term`, use lastComment if given, otherwise push lastComment and entry.

Once we add top-level whitespace with https://github.com/projectfluent/fluent/issues/116, does one of the two become more straightforward? Would we fold that top-level entry into getEntryOrJunk?

Reply:
I think I like the `lastComment` approach better. I'll implement it in a separate commit so that we can draw inspiration from the `entries.push(entry, next)` approach in the future, if needed.